native/core/src/activities.rs

```diff
@@ -1,55 +1,624 @@
-use matrix_sdk::ruma::OwnedEventId;
+use chrono::{NaiveDate, NaiveTime, Utc};
+use matrix_sdk::ruma::{
+    events::room::{create::RoomCreateEventContent, message::TextMessageEventContent},
+    OwnedEventId,
+};
+use object::ActivityObject;
+use urlencoding::encode;
 
 use crate::{
     client::CoreClient,
-    models::{status::membership::MembershipChange, ActerSupportedRoomStatusEvents, AnyActerModel},
+    events::{
+        attachments::AttachmentContent, news::NewsContent, rsvp::RsvpStatus, RefDetails,
+        UtcDateTime,
+    },
+    models::{
+        status::membership::MembershipChange, ActerModel, ActerSupportedRoomStatusEvents,
+        AnyActerModel, EventMeta, Task,
+    },
+    store::Store,
 };
 
+pub mod object;
 pub mod status;
 
 #[derive(Clone, Debug)]
-pub enum Activity {
+pub enum ActivityContent {
     MembershipChange(MembershipChange),
+    RoomCreate(RoomCreateEventContent),
+    RoomName(String),
+    Boost {
```

Comment:
this and the following are essentially moved over from the `NotificationItemInner` but cleaned up (as we can be sure we have some activity object, and we have the other data in the meta info instead)